src/super_context.ml

```diff
@@ -304,12 +304,13 @@ module Libs = struct
       List.map (Preprocess_map.pps preprocess) ~f:Pp.to_string
     in
     let vrequires = vrequires t ~dir ~item in
+    let scope = Lib_db.find_scope' t.libs ~dir in
```

Comment:
Shouldn't we already know the scope wherever `real_requires` is called?

Reply:
We know the scope but we still need a dir to make sure that we depend/generate the .requires files. I could pass the scope along as well, but I thought if I just pass the dir, I'm making it impossible for dir and the scope to diverge here.

Reply:
That's true, but in my mind they are two different things: one is a dependency path and the other is the current directory for which we are generating rules.

Reply:
So I think we should pass both `dir` and `scope` here